.travis.yml

```diff
@@ -12,21 +12,21 @@ services:
 
 env:
   global:
-    - CONTAINER_JAVA_VER="openjdk-11-jdk"
+    - DOCKER_ORG=zencash
+    - IMAGE_NAME=sc-ci-base
+
+script:
+  - source ci/setup_env.sh
+  - ci/travis_keep_alive.sh
+  - ci/script.sh
+
 jobs:
-    #- name: rust_1.51
-    #  env: CONTAINER_RUST_VER='1.51.0'
-    #- name: rust_1.51_asm
-    #  env: CONTAINER_RUST_VER='1.51.0' RUSTFLAGS='--emit=asm'
+  include:
+#    - name: rust_1.51
+#      env: IMAGE_TAG='bionic_rust-1.51.0_jdk-11_latest' TESTS='000,001,002,003,004,005,006'
+#    - name: rust_1.51_asm
+#      env: IMAGE_TAG='bionic_rust-1.51.0_jdk-11_latest' RUSTFLAGS='--emit=asm' TESTS='000,001,002,003,004,005,006'
     - name: rust_1.51_asm_target-feature
-      env: CONTAINER_RUST_VER='1.51.0' RUSTFLAGS='-C target-feature=+bmi2,+adx --emit=asm'
-    #- name: rust_nightly
-    #  env: CONTAINER_RUST_VER='nightly-2021-04-25' RUSTFLAGS='-C target-feature=+bmi2,+adx --emit=asm'
-
-script: >-
-  source ci/setup_env.sh &&
-  docker build --pull --no-cache -t zencash/zendoo-sc-cryptolib-builder ./ci &&
-  bash -c "docker run --rm -v $(pwd):/build -v ${HOME}/key.asc:/key.asc --tmpfs /tmp:uid=$(id -u),gid=$(id -g),exec,mode=1777 \
-    --tmpfs /run:uid=$(id -u),gid=$(id -g),exec,mode=1777 -e RUSTFLAGS -e LOCAL_USER_ID=$(id -u) -e LOCAL_GRP_ID=$(id -g) -e CARGO_AUDIT_EXIT_ON_ERROR \
-    $(env | grep -E '^CONTAINER_' | sed -n '/^[^\t]/s/=.*//p' | sed '/^$/d' | sed 's/^/-e /g' | tr '\n' ' ') \
-    zencash/zendoo-sc-cryptolib-builder /build/ci/start_ci.sh"
+      env: IMAGE_TAG='bionic_rust-1.51.0_jdk-11_latest' RUSTFLAGS='-C target-feature=+bmi2,+adx --emit=asm' TESTS='000,001,002,003,004,005,006'
```

Comment:
Having 6 stages is a bit overdoing perhaps ?

Reply:
It is all the commands we had before in one file but split into separate files to mimic the most recent ginger lib architecture: https://github.com/HorizenOfficial/ginger-lib/blob/master/.travis.yml

Reply:
This doesn't change overall execution time by much, maybe 1-2s. But has the advantage of easier identification of the last failed command should there be an error.

Reply:
I understand, but to me it's a bit overdoing having, for instance, a stage for "cargo clean" (a command that never fails and even if it does, it doesn't provide any useful information), and another stage for "cargo build" (that also won't fail if the "cargo test" stage passed)

Reply:
If we don't care about the timing information for each command we can certainly aggregate scripts 000 - 001 or even 000 - 002 into one. On cargo audit it makes sense to have a separate script as that could fail from time to time.
```
Testscript: /build/ci/tests/000-cargo-tests.sh Status: passed Exitcode: 0 Duration: 835.152639906 seconds
Testscript: /build/ci/tests/001-cargo-clean.sh Status: passed Exitcode: 0 Duration: 0.261754856 seconds
Testscript: /build/ci/tests/002-cargo-build.sh Status: passed Exitcode: 0 Duration: 411.551721828 seconds
Testscript: /build/ci/tests/003-cargo-audit.sh Status: passed Exitcode: 0 Duration: 26.128532029 seconds
Testscript: /build/ci/tests/004-mvn-build.sh Status: passed Exitcode: 0 Duration: 24.674652258 seconds
Testscript: /build/ci/tests/005-mvn-tests.sh Status: passed Exitcode: 0 Duration: 191.408469648 seconds
Testscript: /build/ci/tests/006-mvn-publish.sh Status: passed Exitcode: 0 Duration: 0.003049580 seconds
```

Reply:
I agree merging 000 to 002. Probably also 004-006 ?

Reply:
Mergin 000-002